### ISSUE TITLE
Printing FQCN's in console

### ISSUE THREAD

**Issue by bbrala** [rectorphp/rector#8644]
# Feature Request

When reporting on what rectors applied rector prints only the class name, ignoring the namespace. This means there can be collisions on classes if you combine multiple sources of rectors. I could make 2 classes in different namespaces that do pretty much the same but for a different 'version' of a library. These can then collide, which means you do now know what rector was actually applied.

The reason for this is the line: https://github.com/rectorphp/rector/blob/c6998bfca94cd21200dfa1bd9ce18f6ee674515f/src/ChangesReporting/Output/ConsoleOutputFormatter.php#L130

Which just removes ecverything before the class name.

It would be great if it would be possible to print full classes there, this would make is possible to automate against applied rectors. The reason i want to do this is that we want to automate the minimum version of a library (Drupal in this case) that would be needed to support the changes made by rector. 

We have rector that just change the deprecated code, which means your new minimum version is the version the deprecation was removed in, but we also support BC code which means as long as the minimum version is equal to the version that started supporting using a helper class to load deprecated code conditionally you only need that version as minimum.

If we have FQCN it could even reflect on the classes if they implement the interface that allows for BC rector fixes.

So basically:

Could we either get FQCN's in the reporter, or have a way to report FQCN's through a toggle?

**Comment by TomasVotruba** [comment]
Hi, thanks for suggestion.

This is done on purpose, to avoid long report outputs and make classes readable. We also use same convention in [rules overview](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md) and other places.

I want to keep it simple. It also motivates us to keep even short class names unique. There should never be 2 identical short classes. Instead those should be named in unique clear way. Same applies for custom rules.

Thanks for understanding

**Comment by bbrala** [comment]
Fair enough. I'll find a way to work around that.
